### PR TITLE
[V2] fix: fix error handling of attach status updates

### PR DESCRIPTION
**Pull request by edreed**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fixes error handling of `AzVolumeAttachment` attach status updates and improves resiliency to transient network connectivity issues.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by sunpa93** [comment]
Test failures unrelated to changes made in this PR.

**Comment by sunpa93** [comment]
Test failures unrelated to the PR changes.

**Review by sunpa93 — APPROVED** [review]
/lgtm

File: pkg/azureutils/azure_disk_utils.go [hunk 1]
```diff
@@ -1416,7 +1416,24 @@ func isFatalNetError(err error) bool {
...
+
+func isOtherNetError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	msg := err.Error()
+
+	switch {
+	case strings.Contains(msg, "broken pipe"):
```
Comment (mukhoakash) [inline pkg/azureutils/azure_disk_utils.go:1430]: case insensitive string comparison might be better here, Using `strings.ToLower(msg)`?

Comment (edreed) [reply pkg/azureutils/azure_disk_utils.go:1430]: These are hardcoded error strings which are not going to change, so a case-insensitive comparison is unnecessary here. This is consistent with how K8s checks for specific error message, see [IsProbableEOF](https://github.com/kubernetes/apimachinery/blob/v0.27.4/pkg/util/net/http.go#L79).

**Pull request merged**